### PR TITLE
Extra logging utils (and couch request logging)

### PR DESCRIPTION
**Pull request by czue**
This allows you to define a log config with the following minimal setup to log all requests to couch that take longer than 10 seconds (with username, domain, and HQ url).

Goes with https://github.com/dimagi/couchdbkit/pull/10

``` python
LOGGING = {
    'version': 1,
    'disable_existing_loggers': True,
    'formatters': {
        'couch-request-formatter': {
            'format': '[%(username)s:%(domain)s] %(hq_url)s %(method)s %(path)s %(duration)s'
        }
    },
    'filters': {
        'hqcontext': {
            '()': 'corehq.util.log.HQRequestFilter',
        },
        'slow': {
            '()': 'corehq.util.log.SlowRequestFilter',
            'duration': 10,
        }
    },
    'handlers': {
        'couch-request-handler': {
            'level': 'DEBUG',
            'class': 'logging.StreamHandler',
            'formatter': 'couch-request-formatter',
            'filters': ['slow', 'hqcontext'],

        },
    },
    'loggers': {
        'couchdbkit.request': {
            'handlers': ['couch-request-handler'],
            'level': 'DEBUG',
            'propagate': False,
        },
    }
}
```

(here is sample output for the dashboard with no caching and a threshold of `0`)

```
[user@dimagi.com:cory] /a/cory/dashboard/project/ HEAD /hqdev/ 0:00:00.001148
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET _design/domain/_view/domains 0:00:00.003493
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET DOMAIN_MODULE_CONFIG 0:00:00.001166
[user@dimagi.com:cory] /a/cory/dashboard/project/ HEAD /hqdev/ 0:00:00.000706
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-custom_menu_bar 0:00:00.001087
[user@dimagi.com:cory] /a/cory/dashboard/project/ HEAD /hqdev/ 0:00:00.000723
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET _design/app_manager/_view/applications_brief 0:00:00.001693
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET _design/app_manager/_view/applications_brief 0:00:00.002293
[user@dimagi.com:cory] /a/cory/dashboard/project/ HEAD /hqdev/ 0:00:00.001052
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET _design/reportconfig/_view/configs_by_domain 0:00:00.002584
[user@dimagi.com:cory] /a/cory/dashboard/project/ HEAD /hqdev__meta/ 0:00:00.001205
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET _design/userreports/_view/report_configs_by_domain 0:00:00.001775
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-messaging_status 0:00:00.040394
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-demo-reports 0:00:00.001620
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-demo-reports 0:00:00.037239
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-demo-reports 0:00:00.039264
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-demo-reports 0:00:00.039329
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET _design/couchexport/_view/saved_export_schemas 0:00:00.001941
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-bulk_archive_forms 0:00:00.040189
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-revamped_exports 0:00:00.038589
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-revamped_exports 0:00:00.039336
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-locations 0:00:00.001783
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET _design/app_manager/_view/applications_brief 0:00:00.002812
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET hqFeatureToggle-sms_performance_feedback 0:00:00.001243
[user@dimagi.com:cory] /a/cory/dashboard/project/ POST _design/domain/_view/by_status 0:00:00.002613
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET 8f9756be9b1c7f28057d707b405cf6fa 0:00:00.001444
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET 8f9756be9b1c7f28057d707b405cf6fa 0:00:00.001601
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET 8f9756be9b1c7f28057d707b405cf6fa 0:00:00.002020
[user@dimagi.com:cory] /a/cory/dashboard/project/ GET 8f9756be9b1c7f28057d707b405cf6fa 0:00:00.001379
```

### CODE REVIEW

File: corehq/util/log.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 from collections import defaultdict
 from itertools import islice
+from logging import Handler, Filter, LoggerAdapter
```
Comment (dimagimon) [inline corehq/util/log.py]: F401 'Handler' imported but unused

Comment (dimagimon) [inline corehq/util/log.py]: F401 'LoggerAdapter' imported but unused

**Comment by czue** [comment]
@dimagi/team-commcare-hq

**Comment by snopoke** [comment]
nice

**Comment by snopoke** [comment]
Does the couchdbkit version number in requirements need to be bumped?

**Comment by czue** [comment]
@snopoke yeah - wanted to wait till it was merged before publishing. will PR the update after https://github.com/dimagi/couchdbkit/pull/11 is merged

**Pull request merged**